engine/src/main/java/com/datatorrent/stram/webapp/StramWebServices.java

```diff
@@ -492,6 +494,42 @@ public JSONObject getContainer(@PathParam("containerId") String containerId) thr
     return new JSONObject(objectMapper.writeValueAsString(ci));
   }
 
+  @GET
+  @Path(PATH_PHYSICAL_PLAN_CONTAINERS + "/{containerId}" + "/stackTrace")
+  @Produces(MediaType.APPLICATION_JSON)
+  public JSONObject getContainerStackTrace(@PathParam("containerId") String containerId) throws Exception
+  {
+    init();
+
+    if (containerId.equals(System.getenv(ApplicationConstants.Environment.CONTAINER_ID.toString()))) {
+      return new JSONObject(StackTrace.getJsonFormat());
+    }
+
+    StreamingContainerAgent sca = dagManager.getContainerAgent(containerId);
+
+    if (sca == null) {
+      throw new NotFoundException();
+    }
+
+    if (!sca.getContainerInfo().state.equals("ACTIVE")) {
+      throw new RuntimeException("Container is not active");
+    }
+
+    int timeOut = 1000;
```

Comment:
We should probably make the constants 1000 and 10 be configurable, or at least make it a static final variable so that it's easily modified.